magento2-nginx/fpm-7.0/usr/local/share/magento2/magento_functions.sh

```diff
@@ -381,7 +385,41 @@ function do_magento_tail_logs() {
     /app/var/log/system.log
 }
 
+function do_magento_create_admin_user() {
+  set +x
+  if [ -z "${MAGENTO_ADMIN_USERNAME}" ] || [ -z "${MAGENTO_ADMIN_PASSWORD}" ]; then
+    set -x
+    return 0
+  fi
+  local SQL="SELECT 1 FROM admin_user WHERE username='$MAGENTO_ADMIN_USERNAME'"
+  local HAS_ADMIN_USER=0
+  set +e
+  if [ -n "$DATABASE_PASSWORD" ]; then
+    echo "$SQL" | mysql -h"$DATABASE_HOST" -u"$DATABASE_USER" -p"$DATABASE_PASSWORD" "$DATABASE_NAME" | grep -q 1
+    HAS_ADMIN_USER="$?"
+  else
+    echo "$SQL" | mysql -h"$DATABASE_HOST" -u"$DATABASE_USER" "$DATABASE_NAME" | grep -q 1
+    HAS_ADMIN_USER="$?"
+  fi
+  set -e
+
+  if [ "$HAS_ADMIN_USER" != 0 ]; then
+    set +x
+    as_code_owner "bin/magento admin:user:create \
+      --admin-user='${MAGENTO_ADMIN_USERNAME}' \
+      --admin-password='${MAGENTO_ADMIN_PASSWORD}' \
+      --admin-email='admin@example.com' \
+      --admin-firstname='Development' \
+      --admin-lastname='Admin'"
+    set -x
+  fi
+}
+
 function do_magento2_build() {
+  if [ ! -f '/app/composer.json' ]; then
```

Comment:
what's the significance of there not being a composer.json? A comment would be best if significant reason to